classes/completion/custom_completion.php

```diff
@@ -0,0 +1,65 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace mod_subcourse\completion;
+
+/**
+ * Custom completion rules for mod_subcourse
+ *
+ * @package     mod_subcourse
+ * @copyright   2022, ISB Bayern
+ * @author      Stefan Hanauska <stefan.hanauska@csg-in.de>
+ * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class custom_completion extends \core_completion\activity_custom_completion {
```

Comment:
Is there some deprecation message displayed somewhere or so?

Reply:
Deprecation comes from https://github.com/moodle/moodle/blob/master/lib/completionlib.php#L727 (since 3.11 I think)